### PR TITLE
Make AnalyticsOptIn use its own scoped pref store

### PR DESCRIPTION
**Pull request by chanseokoh**
Make Analytics preferences use its own scoped preference store path: `configuration://com.google.cloud.tools.eclipse.usagetracker`. It doesn't make much sense to me that it uses the bundle default store (which is by default instance-scoped if I hadn't overridden.)

I've removed the `Activator` and the awkward method override, which were the results of the past when I was completely new to Eclipse.

One thing I noticed is that the initializer is called only when the store path matches the bundle name (`com.google.cloud.tools.eclipse.preferences`), so there seems no way to do `setDefault()` on `configuration://com.google.cloud.tools.eclipse.usagetracker`.

### CODE REVIEW

**Comment by briandealwis** [comment]
Would it make sense to move all of the Analytics stuff out of .preferences into .usagetracker?

File: plugins/com.google.cloud.tools.eclipse.preferences/plugin.xml [hunk 1]
```diff
@@ -14,18 +14,12 @@
       </page>
    </extension>
    <extension
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.preferences/plugin.xml]: Would it make sense to move this to .usagetracker?

Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.preferences/plugin.xml]: You mean including everything Analytics-related, such as `AnalyticsOptInArea.java` and `AnalyticsPreferences.java` in this bundle, right?

Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.preferences/plugin.xml]: Oh right you were fetching the value directly via ConfigurationScope.  It makes more sense if we were using the IPreferenceService, which will look up a set of scopes (default: instance, configuration, then default scope).

Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.preferences/plugin.xml]: > You mean including everything Analytics-related, such as AnalyticsOptInArea.java and AnalyticsPreferences.java in this bundle, right?

Yeah.

**Comment by chanseokoh** [comment]
Let me check if I can use an initializer after moving code.

**Comment by chanseokoh** [comment]
Yeah, it's only by `AnalyticsPingManager`, and it felt like adding more boilerplate code with little benefit.

**Pull request merged**